sunpy/timeseries/metadata.py

```diff
@@ -0,0 +1,515 @@
+            end   = metatuple[0].end
+            out_of_range = False
+            
+            # Find truncations
+            if start < timerange.start and end > timerange.start:
+                # Truncate the start
+                start = timerange.start
+            elif start > timerange.end:
+                # Metadata time range starts after truncated data ends.
+                out_of_range = True
+            if end > timerange.end and start < timerange.end:
+                # Truncate the end
+                end = timerange.end
+            elif end < timerange.start:
+                # Metadata time range finishes before truncated data starts.
+                out_of_range = True
+            
+            # Add the values if applicable
+            if not out_of_range:
+                truncated.append((TimeRange(start, end), metatuple[1], metatuple[2]))
+                
+        # Update the original list
+        self.metadata = truncated
+
+    @property
+    def columns(self):
+        """Returns a list of all the names of the columns in the metadata."""
+        all_cols = set()
+        for metatuple in self.metadata:
+            all_cols.update(metatuple[1])
+        return list(all_cols)
+        
+    def remove_columns(self, colnames):
+        """Removes the given column/s from the TimeSeriesMetaData object.
+       
+        Parameters
+        ----------
+        colnames : `str` or `list`
+            The name or names of the columns to be removed.
+        """
+        # Parameters
+        if isinstance(colnames, str):
+            colnames = [ colnames ]
+
+        # Create a new list with all metadata entries without colnames
+        reduced = []
+        for metatuple in self.metadata:
+            # Check each colname
+            for colname in colnames:
+                if colname in metatuple[1]:
+                    # Removed from the list.
+                    metatuple[1].remove(colname)
+            # Add the column if it still has some columns listed
+            if len(metatuple[1]) > 0:
+                reduced.append(metatuple)
+
+        # Update the original list
+        self.metadata = reduced
+
+
+
+    def rename_column(self, old, new):
+        """
+        Change the name of a column in all the metadata entries.
+        
+        Parameters
+        ----------
+        old : `str`
+            The original column name to be changed.
+
+        new : `str`
+            The new column name.
+        """
+        for i in range(0, len(self.metadata)):
+            # Update the colnames
+            colnames = self.metadata[i][1]
+            colnames = [w.replace(old, new) for w in colnames]
+            
+            # Replace values
+            self.metadata[i] = ( self.metadata[i][0], colnames, self.metadata[i][2] )
+
+    def _validate_meta(self, meta):
+        """
+        Validate a meta argument.
+        """
+        return True
+
+    def _to_string(self, depth=10, width=99):

```

Comment:
due to the customisation of this it should probably not be hidden, i.e. not start with `_`